src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -132,6 +158,225 @@ internal void FinalizeDebugInfo()
             }
         }
 
+        /// <summary>
+        /// If <see cref="DebugFlag"/> is set to true, emits a location allowing for a breakpoint when debugging. Expects a 0-based <see cref="Position"/>
+        /// relative to the namespace and statement stack of parent <see cref="QsStatement"/>s,
+        /// which is converted to a 1-based <see cref="DebugPosition"/>.
+        /// </summary>
+        /// <param name="relativePosition">The 0-based <see cref="Position"/> at which to emit the location
+        /// relative to the namespace and stack of parent <see cref="QsStatement"/>s.</param>
+        internal void EmitLocationWithOffset(Position? relativePosition)
+        {
+            if (this.DebugFlag)
+            {
+                DISubProgram? sp = this.sharedState.CurrentFunction?.DISubProgram;
+                QsLocation? namespaceLoc = this.sharedState.DIManager.CurrentNamespaceElementLocation;
+
+                if (namespaceLoc != null && relativePosition != null && sp != null)
+                {
+                    Position absolutePosition = namespaceLoc.Offset + this.TotalOffsetFromStatements() + relativePosition;
+                    this.EmitLocation(absolutePosition, sp);
+                }
+            }
+        }
+
+        /// <summary>
+        /// If <see cref="DebugFlag"/> is set to true, emits the current location allowing for a breakpoint when debugging.
+        /// </summary>
+        internal void EmitLocation()
+        {
+            if (this.DebugFlag)
+            {
+                this.EmitLocationWithOffset(Position.Zero);
+            }
+        }
+
+        /// <summary>
+        /// If the debug flag is set to true, this creates the debug information for a local variable.
+        /// </summary>
+        /// <param name="name">The name of the local variable.</param>
+        /// <param name="value">The value of the variable.</param>
+        internal void CreateLocalVariable(string name, IValue value)
+        {
+            if (this.DebugFlag)
+            {
+                // get the DebugInfoBuilder for this variable
```

Comment:
Nitpick (not blocking): In this codebase in general, the first letter in a comment is capitalized.